src/audio/crossover/crossover.c

```diff
@@ -642,215 +562,180 @@ static int crossover_copy(struct comp_dev *dev)
 	else
 		num_sinks = num_assigned_sinks;
 
-	/* Find the number of frames to copy over */
-	for (i = 0; i < num_sinks; i++) {
-		if (!sinks[i])
-			continue;
-		/*
-		 * WARNING: if a different thread happens to lock the same
-		 * buffers in different order, they can deadlock
-		 */
-		sinks_c[i] = buffer_acquire(sinks[i]);
-		avail = audio_stream_avail_frames(&source_c->stream,
-						  &sinks_c[i]->stream);
-		frames = MIN(frames, avail);
```

Comment:
This logic seems to be removed by the commit, is it intentional?
It looks important that determines the processed frames with regard to sink capacities in case of overrun.

Reply:
Same as Curtis asked? The module adapter does this for clients. Though there's also the sinks[i] NULL case if states are not equal. I have not been able to create such test situation, could a NULL sink impact the avail/free. I'm not sure.